File: coffee-system/src/main/java/com/jesperancinha/coffee/api/concurrency/StartupCallable.java
package com.jesperancinha.coffee.api.concurrency;

import java.util.List;
import java.util.concurrent.Callable;

/**
 * Created by joao on 8-5-16.
 */
public interface StartupCallable extends QueueCallable {
    List<Callable<Boolean>> getAllCallables();
}

File: jeorg-kotlin-apps/coffee-paradigms/coffee-system/src/main/kotlin/org/jesperancinha/coffee/system/concurrency/StartupCallable.kt
package org.jesperancinha.coffee.system.concurrency

import org.jesperancinha.coffee.system.api.concurrency.QueueCallable

/**
 * Created by joao on 8-5-16.
 */
class StartupCallable : QueueCallableAbstract(), QueueCallable {
    @Throws(Exception::class)
    override fun call() = true
}